test/utils/utils.go

```diff
@@ -566,3 +582,48 @@ func VerifyCfgMapOrSecretEnvVarExists(c client.Client, expectResource ExpectApiR
 		return false
 	}
 }
+
+func ApplyPrometheusCRD() error {
+	wd, _ := os.Getwd()
+	promCmd := exec.Command("kubectl", "apply", "-f", wd+"/test/configs/prometheus-crd.yaml", "--server-side")
+	if _, err := Run(promCmd); err != nil {
+		return fmt.Errorf("failed to apply Prometheus CRD: %v", err)
+	}
+	cmd := exec.Command("kubectl", "apply", "-f", wd+"/test/configs/rbac.yaml", "--server-side")
+	if _, err := Run(cmd); err != nil {
+		return fmt.Errorf("failed to apply RBAC: %v for Prometheus", err)
+	}
+	return nil
+}
+
+func VerifyServiceMonitor(c client.Client, name, ns string) bool {
+	sm := &monitoringv1.ServiceMonitor{}
+	err := c.Get(context.Background(), client.ObjectKey{
+		Name:      name,
+		Namespace: ns,
+	}, sm)
+	return err == nil
+}
+
+func VerifyServiceMonitorDeleted(c client.Client, name, ns string) bool {
+	sm := &monitoringv1.ServiceMonitor{}
+	err := c.Get(context.Background(), client.ObjectKey{
+		Name:      name,
+		Namespace: ns,
+	}, sm)
+	return errors.IsNotFound(err)
+}
+
+func QueryPrometheus(query string) (*PrometheusResponse, error) {
```

Comment:
Thanks for adding this! Maybe for future, but we can probably copy some e2e thanos methods that do similar, if we want to check metrics.

For now I think this works well!